FirebaseSessions/Sources/SessionGenerator.swift

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+@_implementationOnly import FirebaseInstallations
+
+struct SessionInfo {
+  let sessionId: String
+  let previousSessionId: String?
+  let shouldDispatchEvents: Bool
+
+  init(sessionId: String, previousSessionId: String?, dispatchEvents: Bool) {
+    self.sessionId = sessionId
+    self.previousSessionId = previousSessionId
+    shouldDispatchEvents = dispatchEvents
```

Comment:
Could we name this parameter and property something to do with sampling? Eg. `notSampled` or `samplingDisabled` or something.

Also, this can be written `self.shouldDispatchEvents = shouldDispatchEvents`

Reply:
I thought about that and I'm not too particular. I preferred "dispatchEvents" mainly since that is the user facing behavior and sampling is an internal behavior of the session. If you are too particular, happy to rename it.